src/main/java/seedu/uninurse/model/task/RecurringTask.java

```diff
@@ -114,6 +120,46 @@ public boolean equals(Object other) {
 
         return this.getTaskDescription().equals(o.getTaskDescription())
                 && this.getDateTime().equals(o.getDateTime())
-                && this.recurrence.equals(o.recurrence);
+                && this.isSimilarRecurrence(recurrence, o.recurrence);
+    }
+
+    private boolean isSimilarRecurrence(Recurrence recurrence1, Recurrence recurrence2) {
+        if (recurrence1.toString().equals(recurrence2.toString())) {
+            return true; // if both are equal
+        }
+        if ((recurrence1.toString() + "S").equals(recurrence2.toString())) {
```

Comment:
Given the new changes (e.g. `2 day` input taken as `2 days` internally, would this check be redundant?

Reply:
'Izzah pointed out a good point and I also think that this check is possibly redundant.